src/index.js

```diff
@@ -160,44 +158,40 @@ module.exports = class WyzeAPI {
 
     const config = {
       baseURL: this.authBaseUrl,
-      headers: { "x-api-key": this.authApiKey, "User-Agent": this.userAgent },
-    };
-
-    if (this.apiKey && this.keyId) {
-      url = "api/user/login";
-      config.headers = {
+      headers: {
+        "x-api-key": this.authApiKey,
         apikey: this.apiKey,
         keyid: this.keyId,
         "User-Agent": this.userAgent,
-      };
-    }
+      },
+    };
 
     return this._performRequest(url, data, config);
   }
 
   async login() {
-    let result = await this._performLoginRequest();
-
-    // Do we need to perform a 2-factor login?
-    if (!result.data.access_token && result.data.mfa_details) {
-      if (!this.mfaCode) {
+    let result;
+    // Do we need apiKey or keyId?
+    if (this.apiKey == null) {
+      throw new Error(
+        'ApiKey Required, Please provide the "apiKey" parameter in config.json'
+      );
+    } else if (this.keyId == null) {
+      throw new Error(
+        'KeyId Required, Please provide the "keyid" parameter in config.json'
+      );
+    } else {
+      try {
+        result = await this._performLoginRequest();
+        if (this.apiLogEnabled)
+          this.log.debug("Successfully logged into Wyze API");
+        await this._updateTokens(result.data);
+      } catch (error) {
```

Comment:
Do we want to log the specific error in the error message here, provide a hint for the user to track down the issue?

Reply:
That would be good.